aopy/visualization.py

```diff
@@ -167,6 +194,104 @@ def plot_spatial_map(data_map, x, y, ax=None, cmap='bwr'):
     ax.set_xlabel('x position')
     ax.set_ylabel('y position')
 
+
+
+def plot_rastor(data, plot_cue, cue_bin, ax):
+    '''
+       Create a rastor plot of neural data
+
+       Args:
+           data (n_trials, n_neurons, n_timebins): neural spiking data (not spike count- must contain only 0 or 1) in the form of a three dimensional matrix
+           plot_cue : If plot_cue is true, a vertical line showing when this event happens is plotted in the rastor plot
+           cue_bin : time bin at which an event occurs. For example: Go Cue or Leave center
+            ax: axis to plot rastor plot
+       Returns:
+           rastor plot in appropriate axis
+    '''
+    n_trial = np.shape(data)[0]
+    n_neurons = np.shape(data)[1]
+    n_bins = np.shape(data)[2]
+
+    color_palette = sns.set_palette("Accent", n_neurons)
+    for n in range(n_neurons):  # set this to 1 if we need rastor plot for only one neuron
+        for tr in range(n_trial):
+            for t in range(n_bins):
+                if data[n, tr, t] == 1:
+                    x1 = [tr, tr + 1]
+                    x2 = [t, t]
+                    ax.plot(x2, x1, color=color_palette(n))
```

Comment:
this seems like it will be very slow. have you looked at https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.eventplot.html